### ISSUE TITLE
Allow to perform registration while keeping the depth buffer

### ISSUE THREAD

**Issue by SirLynix** [OpenKinect/libfreenect#649]
Hi, author of [obs-kinect](https://github.com/SirLynix/obs-kinect) here.

First of all, thank you for your amazing job on libfreenect, I've been using it for my plugin but there's one issue that remains, I need to be able to get registered depth (color-mapped depth as I call it) and regular depth at the same time, however libfreenect seems to be the only kinect backend which doesn't provide a function to map depth to rgb space (libfreenect2 has `Registration::apply`, Kinect SDKs have `MapColorFrameToDepthFrame`, Azure SDK has [k4a_transformation_depth_image_to_color_camera](https://microsoft.github.io/Azure-Kinect-Sensor-SDK/refs/pull/1085/merge/group___functions_gafacffb5f781a9c2df30d4a16241cd514.html#gafacffb5f781a9c2df30d4a16241cd514)).

I see there's the internal `freenect_apply_depth_to_mm` which seems to do what I want, is there a way it could be exposed?

Thank you.

**Comment by piedar** [comment]
I think you might want `freenect_apply_registration` because it aligns the result to the RGB image, while `freenect_apply_depth_to_mm` converts to mm but does not align.

**Comment by SirLynix** [comment]
Yes indeed. I made a fork of libfreeenect to [expose such features](https://github.com/OpenKinect/libfreenect/compare/master...SirLynix:master) which [allowed me to have both depth and color-mapped depth images](https://github.com/SirLynix/obs-kinect/commit/5e67696e7ebe8ec0839899268d12d1e3cf2390b1).

**Comment by piedar** [comment]
Nice!  I had a [quick go](https://github.com/OpenKinect/libfreenect/compare/public-registration) at it the other day.  It's a bit more complicated, but I kept `convert_packed_to_16bit` internal and my registration function handles all the depth modes rather than just `FREENECT_DEPTH_11BIT`.  Let me know what you think.

**Comment by SirLynix** [comment]
Looks good! I will give it a try to check if it works as expected soon, thank you!

**Comment by edugimeno** [comment]
Please please forgive me for hijacking this thread for this question, but there's absolutely no place / forum I can find to place a couple questions about this project. The wiki links to google groups which were closed 4 years ago. There's no other trace of forum for OpenKinect, can you please help me with directions for any currently working forum for OpenKinect / Libfreenect?

Thanks in advance and sorry

**Comment by piedar** [comment]
@edugimeno Alas, there isn't really a forum that I know of.  I check in here on GitHub occasionally to fix bugs and give suggestions, but most contributors have moved on to other projects.
